executor/infoschema_reader.go

```diff
@@ -891,10 +891,19 @@ ForColumnsTag:
 		var columnDefault interface{}
 		if columnDesc.DefaultValue != nil {
 			columnDefault = fmt.Sprintf("%v", columnDesc.DefaultValue)
-			if ft.GetType() == mysql.TypeBit {
-				defaultStr := fmt.Sprintf("%v", columnDesc.DefaultValue)
-				defaultValBinaryLiteral := types.BinaryLiteral(defaultStr)
-				columnDefault = defaultValBinaryLiteral.ToBitLiteralString(true)
+			switch col.GetDefaultValue() {
+			case "CURRENT_TIMESTAMP":
+			default:
+				if ft.GetType() == mysql.TypeTimestamp && columnDefault != types.ZeroDatetimeStr {
+					timeValue, err := table.GetColDefaultValue(sctx, col)
+					if err == nil {
+						columnDefault = timeValue.GetMysqlTime().String()
+					}
+				}
+				if ft.GetType() == mysql.TypeBit {
+					defaultValBinaryLiteral := types.BinaryLiteral(columnDefault.(string))
+					columnDefault = defaultValBinaryLiteral.ToBitLiteralString(true)
+				}
```

Comment:
Perhaps this should be tuned to `DefaultExpr` as well:

```sql
mysql> create table t(a bit default (rand()));
Query OK, 0 rows affected (0.05 sec)

mysql> select column_default from information_schema.columns where TABLE_NAME='t' and TABLE_SCHEMA='test';
+----------------+
| COLUMN_DEFAULT |
+----------------+
| rand()         |
+----------------+
1 row in set (0.02 sec)
```

Reply:
forgot it, fixed. PTAL
@tangenta